### PR TITLE
prometheus: 2.49.0 -> 2.49.1

### LINKED ISSUE #13401: 2.49.0: Change to Accept header sends wrong q= value, causing e.g. RabbitMQ native endpoint to throw HTTP 400
### What did you do?

I upgraded Prometheus from 2.48.1 to the recently released 2.49.0 without any changes to the config.

### What did you expect to see?

I expected all scraping targets to work successfully (as there seem to be no breaking changes in the release notes).

### What did you see instead? Under which circumstances?

Scraping the native Prometheus endpoint of RabbitMQ 3.12 (https://www.rabbitmq.com/prometheus.html#overview-prometheus) ends up in HTTP 400s:

```
HTTP/1.1 400 Bad Request
content-length: 0
date: Mon, 15 Jan 2024 14:46:23 GMT
server: Cowboy
```

I then sniffed the traffic and compared the request Prometheus sent before (2.48.1) and after (2.49.0) and saw the following differences:

```
< User-Agent: Prometheus/2.48.1
< Accept: application/openmetrics-text;version=1.0.0,application/openmetrics-text;version=0.0.1;q=0.75,text/plain;version=0.0.4;q=0.5,*/*;q=0.1
---
> User-Agent: Prometheus/2.49.0
> Accept: application/openmetrics-text;version=1.0.0;q=0.5,application/openmetrics-text;version=0.0.1;q=0.4,text/plain;version=0.0.4;q=0.3,*/*;q=2
```

### System information

Linux 6.2.0-39-generic x86_64

### Prometheus version

```text
prometheus, version 2.48.1 (branch: HEAD, revision: 63894216648f0d6be310c9d16fb48293c45c9310)
  build user:       root@71f108ff5632
  build date:       20231208-23:33:22
  go version:       go1.21.5
  platform:         linux/amd64
  tags:             netgo,builtinassets,stringlabels


vs.


prometheus, version 2.49.0 (branch: HEAD, revision: 683a46cc9999dedf66dd0a4c77644b8d0031d063)
  build user:       root@a83669167414
  build date:       20240115-10:51:05
  go version:       go1.21.6
  platform:         linux/amd64
  tags:             netgo,builtinassets,stringlabels
```

### Prometheus configuration file

_No response_

### Alertmanager version

_No response_

### Alertmanager configuration file

_No response_

### Logs

```text
prometheus[633395]: ts=2024-01-15T14:21:13.219Z caller=scrape.go:1274 level=debug component="scrape manager" scrape_pool=rabbitmq target=http://REDACTED:6666/metrics msg="Scrape failed" err="server returned HTTP status 400 Bad Request"
```

### PR DESCRIPTION
**Pull request by hacklschorsch**
Diff: https://github.com/prometheus/prometheus/compare/v2.49.0...v2.49.1
Changelog: https://github.com/prometheus/prometheus/blob/v2.49.1/CHANGELOG.md

## Description of changes

Only one hour after the Prometheus release 2.49.0, hotfix 2.49.1 came out to fix [a regression that broke scraping for some targets](https://github.com/prometheus/prometheus/issues/13401).

@mfrw was very quick to fetch the upgrade (thanks!) within that 1 hour window, but must have overlooked the hotfix for a full month.

This PR upgrades nixpkgs Prometheus to the latest hotfix version.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mfrw** [comment]
@hacklschorsch - I actually knew about it but for some reason the tests kept failing on me and I did not turn-around to debug.
I guess, I should have created an Issue for others to look at.
My bad :(

**Comment by hacklschorsch** [comment]
@mfrw No worries.  Strange though that the tests failed - this is a hotfix release, with only one small change (which had no tests).

Now, how does something like this find its way into nixpkgs?  Do I have to do anything, or just chill out and wait for the maintainers to find time to look at it?

**Comment by mfrw** [comment]
> @mfrw No worries. Strange though that the tests failed - this is a hotfix release, with only one small change (which had no tests).

@hacklschorsch I am pretty sure, it was my personal dev-box that might have had issues & not the release itself.


> Now, how does something like this find its way into nixpkgs? Do I have to do anything, or just chill out and wait for the maintainers to find time to look at it?

IMO, our (non-committers) work is done. We might have watch out for review items if any (unlikely in this case); other than that, it would be now the maintainers and committers that actually merge this in.

Thank you for the hotfix PR :)

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 288817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>prometheus</li>
    <li>prometheus.cli</li>
    <li>prometheus.doc</li>
  </ul>
</details>

**Review by mfrw — APPROVED** [review]

**Review by fpletz — APPROVED** [review]

**Comment by hacklschorsch** [comment]
Thanks!

**Comment by hacklschorsch** [comment]
Can I expedite this fix's way into stable?

IIUC I can't issue a request for automated backporting, right?  I'd have to add the `backport 23.11` label IIANM?

Can someone add the label to get this fix ported to current stable 23.11?

**Comment by hacklschorsch** [comment]
*nudge nudge* ?  I'd like to upgrade to current nixos-23.11 (for the recent GitLab security patches), but I'm blocked by this missing Prometheus Hotfix in nixos-23.11.

Would be amazing if s.o. could add the backport label for this.  Pretty please? @fpletz @WilliButz @Frostman

**Pull request merged**
